tools/cli-core/src/util.ts

```diff
@@ -170,3 +177,44 @@ export const generateJSONFile = ( filePath: string, data: unknown ) => {
 	);
 	return writeFile( filePath, json );
 };
+
+/**
+ * Get plugin data from main PHP file in JSON format.
+ *
+ * @param {string} tmpRepoPath    - Path to repo.
+ * @param {string} pathToMainFile - Path to plugin's main PHP file.
+ * @param {string} hashOrBranch   - Hash or branch to checkout.
+ * @return {Object} - plugin data in JSON format.
+ */
+export const getPluginData = async (
+	tmpRepoPath: string,
+	pathToMainFile: string,
+	hashOrBranch: string
+) => {
+	const git = simpleGit( { baseDir: tmpRepoPath } );
+	await git.checkout( [ hashOrBranch ] );
```

Comment:
just a heads up, these lines can be replaced with the helper `await checkoutRef(hashOrBranch)`

Reply:
Fixed in https://github.com/woocommerce/woocommerce/pull/34951/commits/4479b1b6ceb9a001b27d37370690202eacde728c